web/src/components/storage/VolumeLocationDialog.jsx

```diff
@@ -0,0 +1,184 @@
+      <label htmlFor={id}>{children}</label>
+    </>
+  );
+};
+
+/**
+ * Renders a dialog that allows the user to change the location of a volume.
+ * @component
+ *
+ * @param {object} props
+ * @param {Volume} props.volume - Volume to edit.
+ * @param {StorageDevice[]} props.devices - Devices available for installation.
+ * @param {string} props.target - Installation target.
+ * @param {StorageDevice|undefined} props.targetDevice - Device selected for installation, if target is a disk.
+ * @param {boolean} [props.isOpen=false] - Whether the dialog is visible or not.
+ * @param {() => void} props.onCancel
+ * @param {(volume: Volume) => void} props.onAccept
+ */
+export default function VolumeLocationDialog({
+  volume,
+  devices,
+  target,
+  targetDevice: defaultTargetDevice,
+  isOpen,
+  onCancel,
+  onAccept,
+  ...props
+}) {
+  const [locationOption, setLocationOption] = useState(defaultOption(volume.target));
+  const [targetDevice, setTargetDevice] = useState(volume.targetDevice || defaultTargetDevice || devices[0]);
+  const [isDedicatedVG, setIsDedicatedVG] = useState(volume.target === "new_vg");
+
+  const selectAutoOption = () => setLocationOption("auto");
+  const selectDeviceOption = () => setLocationOption("device");
+  const toggleDedicatedVG = (_, value) => setIsDedicatedVG(value);
+
+  const isLocationAuto = locationOption === "auto";
+  const isLocationDevice = locationOption === "device";
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    const newVolume = { ...volume };
+
+    if (isLocationAuto) {
+      newVolume.target = "default";
+      newVolume.targetDevice = undefined;
+    }
+
+    if (isLocationDevice) {
+      newVolume.target = isDedicatedVG ? "new_vg" : "new_partition";
+      newVolume.targetDevice = targetDevice;
+    }
+
+    onAccept(newVolume);
+  };
+
+  const isAcceptDisabled = () => {
+    return isLocationDevice && targetDevice === undefined;
+  };
+
+  const autoText = () => {
+    if (target === "disk" && defaultTargetDevice)
+      return sprintf(_("The filesystem will be allocated as a new partition at the installation \
+disk (%s)."), deviceLabel(defaultTargetDevice));
+
+    if (target === "disk")
+      return _("The filesystem will be allocated as a new partition at the installation disk.");
+
+    return _("The file system will be allocated as a logical volume at the system LVM.");
+  };
+
+  return (
+    <Popup
+      title={sprintf(_("Location for %s file system"), volume.mountPath)}
+      isOpen={isOpen}
+      {...props}
+    >
+      <Form id="volume-location-form" onSubmit={onSubmit}>
+        <fieldset className="stack">
+          <legend className="split">
+            <RadioOption id={LOCATION_AUTO_ID} defaultChecked={isLocationAuto} onChange={selectAutoOption}>
+              {_("Automatic")}
+            </RadioOption>
+          </legend>
+          <div>
+            {autoText()}
+          </div>
+        </fieldset>
+
+        <fieldset className="stack">
```

Comment:
I haven't tested it yet, but looks like the (Firefox) problem described at https://github.com/openSUSE/agama/pull/354 has been solved (see https://bugzilla.mozilla.org/show_bug.cgi?id=1829874)

So, If it's the case, we could disable the fieldset instead of going field by field. Should work natively.